test/project_types/script/test_helpers/fake_metadata_repository.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module TestHelpers
+  class FakeMetadataRepository
+    def initialize(file_location)
+      @cache = {}
+
+      schema_major_version = "1"
+      schema_minor_version = "0"
+      use_msgpack = true
+
+      @cache[file_location] = Script::Layers::Domain::Metadata.new(
+        schema_major_version,
+        schema_minor_version,
+        use_msgpack
+      )
```

Comment:
The fake EP repo exposes a [create_extension_point method](https://github.com/Shopify/shopify-cli/blob/da337506316ec55905c7eab6682858c52dbe1d37/test/project_types/script/test_helpers/fake_extension_point_repository.rb#L11-L13) that populates the cache hash. I saw you ask on slack so I wanted to give you another option 🙂  I think what you have is fine since the metadata is pretty minimal and we don't need to test out different iterations of it, so your call if you want to change it

Reply:
I like that approach. I didn't realize it was ok to have methods in the fake that don't exist in the real 👍